jupyter_server_proxy/handlers.py

```diff
@@ -255,22 +297,62 @@ async def proxy(self, host, port, proxied_path):
             self.set_status(500)
             self.write(str(response.error))
         else:
-            self.set_status(response.code, response.reason)
+            # Represent the original response as a RewritableResponse object.
+            original_response = RewritableResponse(orig_response=response)
+            
+            # The function (or list of functions) which should be applied to modify the
+            # response.
+            rewrite_response = self.rewrite_response
+
+            # If this is a single function, wrap it in a list.
+            if isinstance(rewrite_response, (list, tuple)):
+                rewrite_responses = rewrite_response
+            else:
+                rewrite_responses = [rewrite_response]
+
+            # To be passed on-demand as args to the rewrite_response functions.
+            optional_args_to_rewrite_function = {
```

Comment:
Since these were never in a released version, I think just *request* is enough - I don't think it even needs to be 'orig_request'.